### PR TITLE
Raise RTD doc build warnings as errors

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?
Follow up to #1821 as discussed on slack

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
RTD build - https://readthedocs.org/projects/pytorch-lightning/builds/11039091

File: docs/source/conf.py [hunk 1]
```diff
@@ -423,3 +423,7 @@ def find_source():
 
 """
 coverage_skip_undoc_in_source = True
+
+suppress_warnings = [
+    'epub.unknown_project_files'
```
Comment (Borda) [inline docs/source/conf.py:428]:
```suggestion
#    'epub.unknown_project_files'
```
just to try to find the freezing

**Comment by Borda** [comment]
@awaelchli seems to be failing https://readthedocs.org/projects/pytorch-lightning/builds/11039178/
```
The name of the builder is: epub/home/docs/checkouts/readthedocs.org/user_builds/pytorch-lightning/envs/docs-rtd-warn-as-error/lib/python3.7/site-packages/recommonmark/parser.py:65: UserWarning: Container node skipped: type=document
  warn("Container node skipped: type={0}".format(mdnode.t))

Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/pytorch-lightning/envs/docs-rtd-warn-as-error/lib/python3.7/site-packages/sphinx/cmd/build.py", line 276, in build_main
    app.build(args.force_all, filenames)
  File "/home/docs/checkouts/readthedocs.org/user_builds/pytorch-lightning/envs/docs-rtd-warn-as-error/lib/python3.7/site-packages/sphinx/application.py", line 349, in build
    self.builder.build_update()
  File "/home/docs/checkouts/readthedocs.org/user_builds/pytorch-lightning/envs/docs-rtd-warn-as-error/lib/python3.7/site-packages/sphinx/builders/__init__.py", line 299, in build_update
    len(to_build))
  File "/home/docs/checkouts/readthedocs.org/user_builds/pytorch-lightning/envs/docs-rtd-warn-as-error/lib/python3.7/site-packages/sphinx/builders/__init__.py", line 364, in build
    self.finish()
  File "/home/docs/checkouts/readthedocs.org/user_builds/pytorch-lightning/envs/docs-rtd-warn-as-error/lib/python3.7/site-packages/sphinx/builders/html.py", line 638, in finish
    self.handle_finish()
  File "/home/docs/checkouts/readthedocs.org/user_builds/pytorch-lightning/envs/docs-rtd-warn-as-error/lib/python3.7/site-packages/sphinx/builders/epub3.py", line 81, in handle_finish
    self.build_toc()
  File "/home/docs/checkouts/readthedocs.org/user_builds/pytorch-lightning/envs/docs-rtd-warn-as-error/lib/python3.7/site-packages/sphinx/builders/_epub_base.py", line 705, in build_toc
    self.check_refnodes(refnodes)
  File "/home/docs/checkouts/readthedocs.org/user_builds/pytorch-lightning/envs/docs-rtd-warn-as-error/lib/python3.7/site-packages/sphinx/builders/_epub_base.py", line 211, in check_refnodes
    logger.warning(__('duplicated ToC entry found: %s'), node['refuri'])
  File "/home/docs/.pyenv/versions/3.7.3/lib/python3.7/logging/__init__.py", line 1736, in warning
    self.log(WARNING, msg, *args, **kwargs)
  File "/home/docs/checkouts/readthedocs.org/user_builds/pytorch-lightning/envs/docs-rtd-warn-as-error/lib/python3.7/site-packages/sphinx/util/logging.py", line 124, in log
    super().log(level, msg, *args, **kwargs)
  File "/home/docs/.pyenv/versions/3.7.3/lib/python3.7/logging/__init__.py", line 1768, in log
    self.logger.log(level, msg, *args, **kwargs)
  File "/home/docs/.pyenv/versions/3.7.3/lib/python3.7/logging/__init__.py", line 1449, in log
    self._log(level, msg, args, **kwargs)
  File "/home/docs/.pyenv/versions/3.7.3/lib/python3.7/logging/__init__.py", line 1519, in _log
    self.handle(record)
  File "/home/docs/.pyenv/versions/3.7.3/lib/python3.7/logging/__init__.py", line 1529, in handle
    self.callHandlers(record)
  File "/home/docs/.pyenv/versions/3.7.3/lib/python3.7/logging/__init__.py", line 1591, in callHandlers
    hdlr.handle(record)
  File "/home/docs/.pyenv/versions/3.7.3/lib/python3.7/logging/__init__.py", line 901, in handle
    rv = self.filter(record)
  File "/home/docs/.pyenv/versions/3.7.3/lib/python3.7/logging/__init__.py", line 762, in filter
    result = f.filter(record)
  File "/home/docs/checkouts/readthedocs.org/user_builds/pytorch-lightning/envs/docs-rtd-warn-as-error/lib/python3.7/site-packages/sphinx/util/logging.py", line 406, in filter
    raise SphinxWarning(message)
sphinx.errors.SphinxWarning: duplicated ToC entry found: hooks.xhtml

Warning, treated as error:
duplicated ToC entry found: hooks.xhtml
```
maybe just disable whole epub build as we do not use it anyway...?

**Comment by Borda** [comment]
RTD build - https://readthedocs.org/projects/pytorch-lightning/builds/11039622/

File: .readthedocs.yml [hunk 2]
```diff
@@ -9,15 +9,16 @@ version: 2
...
+  fail_on_warning: true
 
 # Build documentation with MkDocs
 #mkdocs:
 #  configuration: mkdocs.yml
 
 # Optionally build your docs in additional formats such as PDF and ePub
-formats: all
+formats:
+  - html
```
Comment (Borda) [inline .readthedocs.yml:20]:
```suggestion
  - htmlzip
```

Comment (awaelchli) [reply .readthedocs.yml]: just saw it now, thanks

**Comment by Borda** [comment]
RTD build - https://readthedocs.org/projects/pytorch-lightning/builds/11039662/

File: docs/source/conf.py (hunk 1 above)
Comment (awaelchli) [reply docs/source/conf.py:428]: the build seems fine now, shall we remove the suppress warning now?

**Comment by awaelchli** [comment]
the build seems fine now, shall we remove the suppress warning now?

**Comment by Borda** [comment]
> the build seems fine now, shall we remove the suppress warning now?

true :]

**Pull request merged**